ddtrace/tracer/context.go

```diff
@@ -39,5 +39,19 @@ func SpanFromContext(ctx context.Context) (Span, bool) {
 // is found in the context, it will be used as the parent of the resulting span. If the ChildOf
 // option is passed, the span from context will take precedence over it as the parent span.
 func StartSpanFromContext(ctx context.Context, operationName string, opts ...StartSpanOption) (Span, context.Context) {
-	return internal.GetGlobalTracer().StartSpanFromContext(ctx, operationName, opts...)
```

Comment:
Note that a separate `interface { StartSpanFromContext(...) }` could do the job too if you want to keep this method in `tracer` and call it only in that case. A `startSpanFromContext()` function could hold the logic below, also usable from your method:

```go
type spanStarter interface {
  StartSpanFromContext(ctx context.Context, operationName string, opts ...StartSpanOption) (Span, context.Context)
}

tracer := internal.GetGlobalTracer()
if starter, ok := tracer.( .StartSpanFromContext(ctx, operationName, opts...); ok {
  return starter.startSpanFromContext(ctx, operationName, opts...)
} else {
  return startSpanFromContext(ctx, operationName, opts...)
}
```

But now that I wrote it, maybe it's worst compared to the simple `Context` field you added to the start options :-)

Reply:
Interesting idea. But yeah in practice I think it's going to add more complexity/indirection than this patch. So I think I'd prefer to keep going in the current direction.